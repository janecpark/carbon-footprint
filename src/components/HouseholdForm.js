import React, { useState, useContext } from 'react';
import { useHistory } from 'react-router-dom';
import FormContext from '../FormContext';
const HouseholdForm = () => {
  const { formData, setFormData } = useContext(FormContext);
  const [house, setHouse] = useState(1);
  const history = useHistory();
  const handleChange = (e) => {
    setHouse(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    setFormData((prevState) => ({
      ...prevState,
      input_size: house,
    }));
    history.push('/household3');
  };
  
  return (
    <div className="HouseformForm container">
      <form onSubmit={handleSubmit}>
        <div className="form-group">
          <label htmlFor="household">
            How many people live in your household?
          </label>
          <select
            name="household"
            className="form-control"
            id="household"
            onChange={handleChange}
          >
            <option>1</option>
            <option>2</option>
            <option>3</option>
            <option>4</option>
            <option>5+</option>
          </select>
        </div>

        <button className="btn btn-success">Next</button>
      </form>
    </div>
  );
};

export default HouseholdForm;
