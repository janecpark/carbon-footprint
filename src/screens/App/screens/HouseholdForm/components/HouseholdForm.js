import React, { useState, useContext } from 'react';
import { useHistory } from 'react-router-dom';
import FormContext from '../../../FormContext';
import Slider from 'react-rangeslider';
import 'react-rangeslider/lib/index.css';

/** Shows a form to get information about household  */

const HouseholdForm = () => {
  const { setFormData } = useContext(FormContext);
  const [house, setHouse] = useState(1);
  const history = useHistory();
  const handleChange = (value) => {
    setHouse(value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    setFormData((prevState) => ({
      ...prevState,
      input_size: house || 2,
    }));
    history.push('/household3');
  };

  const labels = {
    1: '1',
    2: '2',
    3: '3',
    4: '4',
    5: '5+',
  };

  return (
    <div className="HouseformForm ">
      <form onSubmit={handleSubmit} className="container">
        <div className="form-group">
          <h4>How many people live in your household?</h4>
          <Slider
            orientation="horizontal"
            onChange={handleChange}
            min={1}
            max={5}
            tooltip={false}
            value={house}
            labels={labels}
          />
        </div>

        <button className="btn btn-success mt-4">Next</button>
      </form>
    </div>
  );
};

export default HouseholdForm;
