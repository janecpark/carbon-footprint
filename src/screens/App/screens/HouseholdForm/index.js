import HouseholdForm from './components/HouseholdForm';

export default HouseholdForm;