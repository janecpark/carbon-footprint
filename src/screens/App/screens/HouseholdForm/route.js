import { routeConstants } from '../../../../shared/constants';
import HouseholdForm from "./";

export default {
    path: routeConstants.HOUSEHOLD.route,
    component: HouseholdForm
};