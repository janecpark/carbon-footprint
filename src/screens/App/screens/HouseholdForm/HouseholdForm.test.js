import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import HouseholdForm from './components/HouseholdForm';
import {
  UserProvider,
  FormProvider,
  ResultProvider,
  ActionProvider,
} from '../testUtils';

it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <HouseholdForm />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
});

it('matches snapshot', function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <HouseholdForm />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


test('renders text on page', () => {
  const { getByText } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <HouseholdForm />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  const element = getByText(/How many/);
  expect(element).toBeInTheDocument();
});
