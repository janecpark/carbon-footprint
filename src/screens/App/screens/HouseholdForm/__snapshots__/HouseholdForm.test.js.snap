// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="HouseformForm "
  >
    <form
      class="container"
    >
      <div
        class="form-group"
      >
        <h4>
          How many people live in your household?
        </h4>
        <div
          aria-orientation="horizontal"
          aria-valuemax="5"
          aria-valuemin="1"
          aria-valuenow="1"
          class="rangeslider rangeslider-horizontal"
        >
          <div
            class="rangeslider__fill"
            style="width: 0px;"
          />
          <div
            class="rangeslider__handle"
            style="left: 0px;"
            tabindex="0"
          >
            <div
              class="rangeslider__handle-label"
            />
          </div>
          <ul
            class="rangeslider__labels"
          >
            <li
              class="rangeslider__label-item"
              data-value="5"
              style="left: 0px;"
            >
              5+
            </li>
            <li
              class="rangeslider__label-item"
              data-value="4"
              style="left: 0px;"
            >
              4
            </li>
            <li
              class="rangeslider__label-item"
              data-value="3"
              style="left: 0px;"
            >
              3
            </li>
            <li
              class="rangeslider__label-item"
              data-value="2"
              style="left: 0px;"
            >
              2
            </li>
            <li
              class="rangeslider__label-item"
              data-value="1"
              style="left: 0px;"
            >
              1
            </li>
          </ul>
        </div>
      </div>
      <button
        class="btn btn-success mt-4"
      >
        Next
      </button>
    </form>
  </div>
</DocumentFragment>
`;
